caikit_nlp/toolkit/text_generation/model_run_utils.py

```diff
@@ -0,0 +1,241 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions used for executing run function for text_generation"""
+
+# Standard
+from typing import Optional, Tuple
+
+# Third Party
+from transformers import StoppingCriteria
+import torch
+
+# First Party
+from caikit.core.data_model.producer import ProducerId
+from caikit.core.toolkit.errors import error_handler
+from caikit.interfaces.nlp.data_model import (
+    GeneratedTextResult,
+    GeneratedTextStreamResult,
+)
+import alog
+
+log = alog.use_channel("RUN_UTILS")
+error = error_handler.get(log)
+
+GENERATE_FUNCTION_ARGS = """
+    text: str
+        Input string to be used to the generation model.
+    max_new_tokens: int
+        The maximum numbers of tokens to generate.
+        Default: 20
+    min_new_tokens: int
+        The minimum numbers of tokens to generate.
+        Default: 0 - means no minimum
+    truncate_input_tokens: int
+        Truncate inputs to provided number of tokens. This can be
+        use to avoid failing due to input being longer than
+        configured limits.
+        Default: 0 - means don't truncate, thus throw error.
+    decoding_method: str
+        Parameters for conditionally penalizing / boosting
+        candidate tokens during decoding.
+        Options: "GREEDY" (default), "SAMPLING"
+    top_k: int
+        The number of highest probability vocabulary tokens to keep for
+        top-k-filtering. Only applicable when decoding_method is SAMPLING.
+        Default: 0 - means disabled
+    top_p: float
+        If set to float < 1, only the smallest set of most probable tokens
+        with probabilities that add up to top_p or higher are kept for
+        generation. Only applicable when decoding_method is SAMPLING.
+        Default: 0.0 - means disabled - equivalent to 1.0
```

Comment:
Is there a source for this? I took a quick glance at the top p logit warper, and don't see 0 being handled as a special case, although I haven't looked closely into how this hooks into generate very recently 

Top P logic warper [source](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py#L388) for reference, very possible I am missing something

Reply:
this was mentioned in the proto file